### PR TITLE
Numerous changes

### PR DESCRIPTION
**Pull request by dstolfa**
This pull request elaborates on the branching instructions, updates pseudo-code to reflect reality more and explains some of the instructions better.

### CODE REVIEW

File: specification/chap-dtrace-intermediate-format.tex [hunk 1]
```diff
@@ -16,7 +16,10 @@ \section{The DIF Interpreter}
 within the \subroutine{dtrace_dif_emulate} subroutine, which has, as
 its first argument, a pointer to a DIF Object or
 \struct{dtrace_difo_t}.  Instructions are executed one at a time,
-until they are exhausted or an error causes interpretation to end.
+until they are exhausted or an error causes interpretation to end. The DIF
+objects are verified in the \subroutine{dtrace_difo_validate} subroutine and
+the DIF interpreter ignores any bounds checking within the
+\subroutine{dtrace_dif_emulate} subroutine for precisely this reason.
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: "precisely this reason"  I think this needs to be reworded or you need to explain the reason.

File: specification/chap-dtrace-intermediate-format.tex [hunk 2]
```diff
@@ -56,14 +59,22 @@ \subsection{Comparison and Test Instructions}
...
 calling DIF program but are only used internally by the interpretation
 routine.
 
 \textbf{XXXRW: Something about instructions}
 
-\textbf{XXXRW: Something about branches}
+\subsection{Branching Instructions}
+
+DIF has eleven branching instructions split into two types: signed and unsigned
+branching instructions. The signed branching instructions take into account
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: s/branching instructions//

File: specification/chap-dtrace-intermediate-format.tex [hunk 3]
```diff
@@ -56,14 +59,22 @@ \subsection{Comparison and Test Instructions}
...
+that the number may be negative, while the unsigned instructions are meant to
+be used with exclusively positive numbers. One thing all of the branching
+instructions have in common is that they load the new \register{pc} register
+from the \registerop{label} field in the Branch Format as will described in
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: as described in (no need to say 'will be')

File: specification/chap-dtrace-intermediate-format.tex [hunk 4]
```diff
@@ -129,8 +140,14 @@ \subsection{Global Variables}
 \label{sec:globals-vars}
 
 Global variables are global to all the clauses in a D script and are
-references to simple names witin the D script.  Space for global
-variables is statically allocated on each invocation of a script.
+references to simple names within the D script.  Space for global
+variables is statically allocated on each invocation of a script. Additionally,
+global variables are identified using the modified register format as
+will be describ
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: s/will be//

File: specification/chap-dtrace-intermediate-format.tex [hunk 5]
```diff
@@ -129,8 +140,14 @@ \subsection{Global Variables}
...
+ed in Subsection~\ref{subsec:r-format} the case of arrays and the
+wide-immediate format which will be described in
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: s/will be//

File: specification/chap-dtrace-intermediate-format.tex [hunk 6]
```diff
@@ -177,7 +194,34 @@ \subsection{Register Format (R-Format)}
   operation
 \end{description}
 
+
+Furthermore, a modified version of the Register Format is used when loading and
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: s/Futhermore//

File: specification/chap-dtrace-intermediate-format.tex [hunk 7]
```diff
@@ -177,7 +194,34 @@ \subsection{Register Format (R-Format)}
...
+storing to array variables in DTrace. The main difference between the regular
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: s/to/data in/

File: specification/chap-dtrace-intermediate-format.tex [hunk 8]
```diff
@@ -197,11 +241,13 @@ \subsection{Branch Format (B-Format)}
...
 
 \subsection{Wide-Immediate Format (W-Format)}
+\label{subsec:w-format}
 
 This format accepts an 8-bit register and 16-bit integer argument (frequently
 an index).
 It is used for a range of instructions including those to load values from
-integer and string constant tables, and to load and store variable values.
+integer and string constant tables, and to load and store variable values that
+are scalars.
```
Comment (gvnn3) [inline specification/chap-dtrace-intermediate-format.tex]: , as well as those that store scalar values in variables.

File: specification/instr/ldga.tex [hunk 9]
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
 
 \subsubsection*{Format}
 
-\textrm{LDGA \%rd, \%r1, \%r2}
+\textrm{LDGA \%rd, var, ndx}
```
Comment (gvnn3) [inline specification/instr/ldga.tex]: We can afford the extra characters, s/ndx/index/

File: specification/instr/ldga.tex [hunk 10]
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
...
 \end{bytefield}
 \end{center}
 
 \subsubsection*{Description}
 
 The \instruction{ldga} instruction looks up the value of a DTrace
-built-in variable based on the value in \registerop{r1} with an
-optional array index in \registerop{r2}.
+built-in variable based on the value in \registerop{var} with an
+optional array index in \registerop{ndx}. \\
```
Comment (gvnn3) [inline specification/instr/ldga.tex]: Why the trailing \\ ?

File: specification/instr/ldga.tex [hunk 11]
```diff
@@ -6,28 +6,31 @@ \subsection*{LDGA: load a DTrace built-in variable}
...
+
+Contrary to  \instruction{ldgs}, the variable identifier is 8 bits long, and
```
Comment (gvnn3) [inline specification/instr/ldga.tex]: s/Contrary to/Unlike the/

File: specification/instr/ldgs.tex [hunk 12]
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
 
 \subsubsection*{Description}
 
-The \instruction{ldgs} instruction looks up a user defined variable in
-a DIF program and places its value into \registerop{rd}.
+The \instruction{ldgs} instruction functions in two ways and is intended to be
```
Comment (gvnn3) [inline specification/instr/ldgs.tex]: s/functions in two ways/has two modes of operation/

File: specification/instr/ldgs.tex [hunk 13]
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
...
+used only for scalar values. The first mode of operation is when the value
+provided in \registerop{var} is lesser than DIF\_VAR\_OTHER\_UBASE. This will
```
Comment (gvnn3) [inline specification/instr/ldgs.tex]: s/lesser/less/

File: specification/instr/ldgs.tex [hunk 14]
```diff
@@ -19,8 +19,17 @@ \subsubsection*{Format}
...
+cause DTrace to look up a pre-defined scalar variable such as curthread, while
+the second mode of operation will result in looking up a user defined variable
+in a DIF program. The result of this instruction will be put into the register
+\registerop{rd}. \\
+
+Contrary to the \instruction{ldga} instruction, the \registerop{var} field is
```
Comment (gvnn3) [inline specification/instr/ldgs.tex]: s/Contrary to/Unlike/

File: specification/instr/stgs.tex [hunk 15]
```diff
@@ -18,17 +18,31 @@ \subsubsection*{Format}
 \end{center}
 
 \subsubsection*{Description}
-The \instruction{stgs} instruct stores a value from the
-\registerop{rd} register into a user defined variable.
+
+Much like \instruction{ldgs}, the instruction \instruction{stgs} operates
```
Comment (gvnn3) [inline specification/instr/stgs.tex]: s/Much like/Similar to/

File: specification/instr/stgs.tex [hunk 16]
```diff
@@ -18,17 +18,31 @@ \subsubsection*{Format}
...
+    var = %rd
 \end{verbatim}
 
 \subsubsection*{Constraints}
 
 \subsubsection*{Failure modes}
 
-This instruction has no run-time failure modes beyond its constraints.
+This instruction will fail if the supplied value in the \registerop{var} field
+is lesser than DIF\_VAR\_OTHER\_UBASE.
```
Comment (gvnn3) [inline specification/instr/stgs.tex]: s/lesser/less/

File: specification/instr/ldga.tex (hunk 10 above)
Comment (dstolfa) [inline specification/instr/ldga.tex]: Makes the pdf look more legible IMO. Can remove if you think it shouldn't be there (same idea is in ldgs).

**Comment by dstolfa** [comment]
I've addressed some of the issues you've brought up. If you believe that the `\\` should be removed, placing the text together, I'll update that as well.

**Pull request merged**